lib/events/dynamoevents/dynamoevents.go

```diff
@@ -890,18 +916,45 @@ func (l *Log) indexExists(tableName, indexName string) (bool, error) {
 	return false, nil
 }
 
-// createV2GSI creates the new global secondary Index and updates
+// createV2GSI creates the new global secondary index if it does not exist and updates
 // the schema to add a string key `date`.
 //
 // This does not remove the old global secondary index.
 // This must be done at a later point in time when all events have been migrated as per RFD 24.
 //
 // Invariants:
-// - The new global secondary index must not exist.
-// - This function must not be called concurrently with itself.
 // - This function must be called before the
 //   backend is considered initialized and the main Teleport process is started.
 func (l *Log) createV2GSI(ctx context.Context) error {
+	if err := backend.AcquireLock(ctx, l.backend, indexV2CreationLock, 5*time.Minute); err != nil {
```

Comment:
It looks that relese is always called thus for readability it would be nice  to have separate function like: 
```
runInLock(ctx, backend, lockKey, fn func() error) {
  if err = lock(); err != nil {
   ...
  }
  if err := fn(); err  != nil {
      //release
     return trace.Aggragate(err, lockErr)
  }
  if err = release(); err = nil {
    ...
  }
  ...
}
```

Reply:
Great idea, migrated to such a design now